### DISCUSSION TITLE
Pooling layers vs aggregation operators

### DISCUSSION THREAD

**Discussion by vakker** [pyg-team/pytorch_geometric#9301] [Q&A] [answered]
I'm a bit confused about the difference between the pooling layers and the aggregation operators.

To me they seem to provide the same functionality, which is aggregating the features from a set of nodes (which can be the whole graph in case of global pooling/aggregation), e.g. [aggr.MeanAggregation](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.aggr.MeanAggregation.html#torch_geometric.nn.aggr.MeanAggregation) does the same as [pool.global_mean_pool](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.pool.global_mean_pool.html#torch_geometric.nn.pool.global_mean_pool). 

Is there any particular theoretical difference, or pooling and aggregation are essentially the same concepts? 
If the latter, then would it make sense to unify them?

Thanks!

**Comment by rusty1s** [comment]
`global_mean_pool` is just the functional variant of `MeanAggregation` (and mostly just exist for backward compatibility reasons by now).

**Reply by agunter7** [reply]
I'm confused on what exactly determines whether something is organized as an Aggregation Operator vs Pooling Layer. Not everything in the Pooling Layers section is purely functional, e.g. [EdgePooling](https://pytorch-geometric.readthedocs.io/en/2.6.1/generated/torch_geometric.nn.pool.EdgePooling.html#torch_geometric.nn.pool.EdgePooling).

Is it just a historical distinction or am I misunderstanding something?

**Comment by vakker** [comment]
Thanks for the reply.
So, in general pooling and aggregation layers seem conceptually the same to me. Is there a fundamental difference?

**Reply by rusty1s** [reply, marked as answer]
Yes, they are basically the same (except for the module vs function difference), similar to `nn.Dropout` and `torch.nn.functional.dropout`.
